tools/azsdk-cli/docs/specs/6-package-updater.spec.md

```diff
@@ -0,0 +1,669 @@
+
+- Partially automated in ad-hoc scripts per language / repo.
+- Inconsistent between management-plane and data-plane SDKs (especially version bump timing and changelog handling).
+- Error-prone when performed manually (missed version increments, stale README examples, inconsistent metadata presented across different files, like `pom.xml`, `_metadata.json`).
+- A barrier for integrating an AI agent workflow because required parameters and step order are not encapsulated as composable tools.
+
+### Why This Matters
+
+Without a unified tool layer:
+
+- Contributors spend time re-learning differing per-language conventions.
+- CI or release pipelines fail late due to missing metadata or mismatched versions.
+- Agents cannot reliably assist because state transitions ("now bump version", "now update changelog") aren't formalized.
+- Management-plane vs data-plane discrepancies introduce confusion (e.g., when to bump versions or generate changelog entries).
+
+Formalizing Stage 6 as cohesive tools improves reliability, supports automation-first workflows, and prepares for future LLM-assisted tasks.
+
+---
+
+## Goals and Exceptions/Limitations
+
+### Goals
+
+- [ ] Provide four composable, idempotent tools exposed via CLI and MCP agent: CI Update, Changelog Update, Version Update, Metadata Update.
+- [ ] Normalize execution order and expected outputs across all supported languages.
+- [ ] Encode management-plane vs data-plane behavioral differences explicitly (especially version & changelog logic).
+- [ ] Emit structured JSON results (status + message + next steps hints) enabling orchestration by AI agents and pipelines.
+- [ ] Support configuration-driven per-language scripting (path or inline) without recompiling the tool.
+
+### Exceptions
+
+| Exception | Description | Impact | Workaround |
+|---|---|---|---|
+| Data‑Plane Deferred Version Bump | Data‑plane packages typically defer version adjustment and release‑date updates until the release‑preparedness phase. | The `update-version` tool may be a no‑op for data‑plane flows during Stage 6. | Keep version changes manually for milestone 1. |
+| Changelog Auto‑Generation (Data‑Plane) | Automatic changelog generation is not performed for data‑plane packages; manual editing is expected. | The `update-changelog` tool should not auto‑generate entries for data‑plane packages and should instead surface guidance. | Return a `noop` result with `next_steps` prompting manual edits. Consider future LLM integration to automate this post milestone 1. |
+
+### Language-Specific Limitations
+
+| Language   | Limitation | Impact | Workaround |
+|------------|------------|--------|------------|
+| JavaScript | Code formatting isn't hooked up to the emitter | Noisy diffs or linter failures | Run code formatting step in the `update-metadata.ps1` script |
```

Comment:
Are there any issues with waiting till the Package Metadata & Docs Updates stage to run this code formatting? 
Will needed formatting changes force users to have to go back to the TypeSpec/Code customizations, testing, samples, or validation phase?